com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -201,18 +204,16 @@ private boolean shouldRecreate(CloudServiceExtended service, CloudService existi
             return false;
         }
         ServiceOperation lastOperation = getLastOperation(serviceInstanceEntity);
-        if (lastOperation != null && lastOperation.getType() == ServiceOperationType.CREATE
+        if (lastOperation != null
+            && (lastOperation.getType() == ServiceOperationType.CREATE || lastOperation.getType() == ServiceOperationType.DELETE)
```

Comment:
What about a failed previous update?

Reply:
This check should be remain only for create failed operation state.